jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ProjectAccountSettingsManager.kt

```diff
@@ -42,14 +43,16 @@ abstract class ProjectAccountSettingsManager(private val project: Project) : Sim
     protected val recentlyUsedRegions = MRUList<String>(MAX_HISTORY)
 
     // Internal state is visible for AwsSettingsPanel and ChangeAccountSettingsActionGroup
-    internal var selectedCredentials: ToolkitCredentialsProvider? = null
+    internal var selectedCredentialIdentifier: ToolkitCredentialsIdentifier? = null
     internal var selectedRegion: AwsRegion? = null
 
+    private var selectedCredentialsProvider: ToolkitCredentialsProvider? = null
```

Comment:
Why do we store the provider and the identifier? shouldn't we be able to store just the provider?

Reply:
I believe you are correct, will double check

Reply:
If we dont have the region yet, we cant get the provider.

Reply:
Makes sense, can you add a comment about it? on first glance it seems like duplication

Reply:
Yep, and next small rev i cleaned up it a bit.